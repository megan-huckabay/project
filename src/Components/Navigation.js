// src/components/Navigation.js
import React from 'react';
import { Link } from 'react-router-dom';
import "../Styles/Navigation.css";

const Navigation = () => {
  return (
    <nav>
      <ul>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/About">About</Link>
        </li>
        <li>
          <Link to="/Education">Education</Link>
        </li>
        <li>
          <Link to="/PastExperience">Experience</Link>
        </li>
      </ul>
    </nav>
  );
};

export default Navigation;
