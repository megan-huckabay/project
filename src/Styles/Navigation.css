/* Styles for the navigation container */
.navigation-container {
  padding: 1rem;
  margin: 1rem;
  text-align: center; 
  font-size: 5rem;
  font-weight: bold;
  
  /* Media query for screens with a maximum width of 768px and adjust for smaller screens */
  @media only screen and (max-width: 768px) {
    font-size: 2.5rem; /* Decrease font size */
    margin: 0.5rem; /* Reduce margin */
    padding: 0.5rem; /* Reduce padding */
  }
}

/* Styles for the unordered list inside the navigation container */
.navigation-container ul {
  /* Removes default list styling */
  list-style-type: none;
  display: flex;
  justify-content: center; 
  padding: 1rem; 
  margin: 1rem; 

  /* Additional styles for smaller screens */
  @media only screen and (max-width: 768px) {
    margin: 0.5rem;
    padding: 0.5rem; /* Remove padding for mobile */
    flex-direction: column;
  }
}

/* Styles for list items inside the navigation container */
.navigation-container li {
  margin: 0.5rem; 
  padding: 0.5rem;
}

/* Styles for anchor (link) elements inside the navigation container */
.navigation-container a {
  color: rgb(0, 0, 0);
  text-decoration: none;
  padding: 2rem;
  margin: 2rem; 
  border: 2px dotted black;
  /* Adds rounded corners to the link */
  border-radius: 5rem;
  /* Adds a semi-transparent white background to the link */
  background: rgba(255, 255, 255, 0.5);
  /* Smooth transition for background color and text color changes */
  transition: background-color 0.3s ease, color 0.3s ease;
  /* Displays links as inline-block elements */
  display: inline-block;

  /* Additional styles for smaller screens */
  @media only screen and (max-width: 768px) {
    padding: 1rem; /* Adjust padding for mobile */
    margin: 1rem; /* Adjust margin for mobile */
  }

}

/* Hover styles for anchor elements inside the navigation container */
.navigation-container a:hover {
  /* Background color and text color changes on hover */
  background-color: rgb(118, 236, 240);
  color: white;
}

