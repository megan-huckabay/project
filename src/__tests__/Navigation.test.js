// __tests__/Navigation.test.js

import 'jest-styled-components';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Navigation from '../Components/Navigation';

test('renders the Navigation component and checks styles', () => {
  const { getByText, container } = render(
    <Router>
      <Navigation />
    </Router>
  );

  // Test component rendering
  const homeLink = getByText(/Home/i);
  const aboutLink = getByText(/About/i);
  const educationLink = getByText(/Education/i);
  const experienceLink = getByText(/Experience/i);
  const contactLink = getByText(/Contact/i);

  expect(homeLink).toBeInTheDocument();
  expect(aboutLink).toBeInTheDocument();
  expect(educationLink).toBeInTheDocument();
  expect(experienceLink).toBeInTheDocument();
  expect(contactLink).toBeInTheDocument();

  // Test styles
  expect(container).toMatchSnapshot();
});
