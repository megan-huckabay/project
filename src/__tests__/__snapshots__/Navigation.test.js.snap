// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Navigation component and checks styles 1`] = `
<div>
  <div
    class="navigation-container"
  >
    <nav>
      <ul>
        <li>
          <a
            class="nav-button"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="nav-button"
            href="/About"
          >
            About
          </a>
        </li>
        <li>
          <a
            class="nav-button"
            href="/Education"
          >
            Education
          </a>
        </li>
        <li>
          <a
            class="nav-button"
            href="/Experience"
          >
            Experience
          </a>
        </li>
        <li>
          <a
            class="nav-button"
            href="/HireMe"
          >
            Hire Me
          </a>
        </li>
        <li>
          <a
            class="nav-button"
            href="/Contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
